source/agora/test/QuorumShuffle.d

```diff
@@ -42,140 +43,81 @@ unittest
+        log.trace(format!"make block %s"(block_height));
+        checkDistance(block_height);
+        log.trace(format!"expect block %s"(block_height));
+        network.expectBlock(block_height);
+        log.trace("Block {}:\n{}\n", block_height, prettify(nodes[0].getBlocksFrom(block_height, 1)[0]));
     }
 
-    // at block height 3 a shuffle should occur
-    makeBlock(3);
-    const quorums_2 = nodes[0].getExpectedQuorums(keys, Height(3));
-    // writeln(quorums_2);  // uncomment to check
-    nodes.enumerate.each!((idx, node) =>
-        retryFor(node.getQuorumConfig() == quorums_2[idx], 5.seconds,
-            format("Node %s has quorum config %s. Expected: %s",
-                idx, node.getQuorumConfig(), quorums_2[idx])));
-
-    checkDistance(6);  // make sure we can create the next 3 blocks
-
-    makeBlock(4);
-    makeBlock(5);
-    makeBlock(6);
-
-    const quorums_3 = nodes[0].getExpectedQuorums(keys, Height(6));
-    // writeln(quorums_3);  // uncomment to check
-    // at block height 6 a shuffle occured
-    nodes.enumerate.each!((idx, node) =>
-        retryFor(node.getQuorumConfig() == quorums_3[idx], 5.seconds,
-            format("Node %s has quorum config %s. Expected: %s",
-                idx, node.getQuorumConfig(), quorums_3[idx])));
-
-    checkDistance(9);  // make sure we can create the next 3 blocks
-    makeBlock(7);
-    makeBlock(8);
-
-    // at block height 10 the enrollments expire so we must re-enroll first.
-    // note that this will not have any effect on the quorum shuffle at height 9,
-    // it will still happen with the preimages at distance 9 and not these new
+    // at block height 20 the enrollments expire so we must re-enroll first.
+    // note that this will not have any effect on the quorum shuffle at height 18,
+    // it will still happen with the preimages at distance 19 and not these new
     // enrollments.
-    foreach (node; nodes)
+    void reEnrollNodes ()
     {
-        Enrollment enroll = node.createEnrollmentData();
-        node.enrollValidator(enroll);
+        foreach (node; nodes)
+        {
+            Enrollment enroll = node.createEnrollmentData();
+            node.enrollValidator(enroll);
+
+            // check enrollment
+            nodes.each!(n =>
+                retryFor(n.getEnrollment(enroll.utxo_key) == enroll, 5.seconds));
+        }
+    }
 
-        // check enrollment
-        nodes.each!(n =>
-            retryFor(n.getEnrollment(enroll.utxo_key) == enroll, 5.seconds));
+    QuorumConfig[] checkQuorum (Height height) {
+        if (height > 0) // if not Genesis block
+        {
+            log.trace(format!"generateBlocks to height %s"(height));
+            if (height % GenesisCycle == 0) { // As cycle is 20 we need to re-enroll every 20 blocks
+                iota(height - block_height - 1).each!(_ => makeBlock());
+                reEnrollNodes();
+                makeBlock();
+            } else
```

Comment:
Please follow the style guide. Braces on their own line.

Reply:
Updated